### PR TITLE
adjust the way dynamic-snip%s work

### PR DESCRIPTION
**Pull request by rfindler**
I think this is an improvement and it avoids a specific brittleness in the way frtime snips are rendered.

### CODE REVIEW

File: frp-snip.rkt [hunk 1]
```diff
@@ -72,94 +72,83 @@
...
     
     (define/override (size-cache-invalid)
       (for-each
        (lambda (s) (send s size-cache-invalid))
        copies))
+
+    (define/override (draw dc x y left top right bottom dx dy draw-caret)
+      (for/fold ([y-offset 0])
+                ([line (in-list current)])
+        (send dc draw-text line x (+ y y-offset))
```
Comment (ghcooper) [inline frp-snip.rkt:129]: Will this work for snips that contain non-textual content (e.g., images)? That feature used to work, though I haven't checked too recently.

Comment (rfindler) [reply frp-snip.rkt:129]: Ah, you're right on both counts (this won't work and it used to)!

**Comment by rfindler** [comment]
Okay, I think the current version accounts for (most) snips. It doesn't support the full range of things that snips can do but it seems to work for numbers and picts at least.

Here are two examples that work (these are more interesting when the drracket window is narrow):
```
(build-list (* 10 (modulo seconds 4)) add1)
```

```
(build-list (* 10 (modulo seconds 4))
              (lambda (x) (+ x 1/2)))
```

This example doesn't work (in this version or in 7.9):
```
(require 2htdp/image)
(ellipse (+ 10 (* 5 (modulo seconds 4))) 10 "solid" "red")
```

and neither do examples I tried for the `pict` library, with a similar error.

Are there other snips that are supposed to work?

**Comment by samth** [comment]
This version of your last example works in 8.0, I haven't tried with your PR:

```
(require (lifted 2htdp/image ellipse))
(ellipse (+ 10 (* 5 (modulo seconds 4))) 10 "solid" "red")
```

**Comment by rfindler** [comment]
It does! As does this variant:

```
(ellipse (+ 10 (* 5 (modulo seconds 4)))
           (+ 10 (* 5 (- 4 (modulo seconds 4))))
           "solid" "red")
```

**Pull request merged**